### PR TITLE
New package: brother-mfc5490cn-cupswrapper-1.1.2

### PR DESCRIPTION
**Pull request by haemka**
CUPSwrapper printer driver for Brother MFC-5490CN

depends on #7034 

[binary source](https://support.brother.com/g/b/downloadend.aspx?c=us&lang=en&prod=mfc5490cn_all&os=128&dlid=dlf006158_000&flang=4&type3=561)

### CODE REVIEW

**Comment by Duncaen** [comment]
Would be easier if related commits are in one PR to allow travis to build the packages.
In general it would be nicer to build the wrapper from source as its gpl licensed.
The other part is proprietary and might not allow redistributing binaries, which would require `restricted=yes` to not add the package to the repository.
In both cases allowed or not allowed to distribute, the package should be in the nonfree repository.

Another issue would be depending on 32bit packages, this is not something void does.
32bit only packages should only build for `i686` and then `xbps-src` creates a `-32bit` package from with with the right dependencies.

**Comment by haemka** [comment]
Thanks for your detailed comment. This really helps, as this is my first PR for void. I had both packages lying around for about a year (in fact there was a PR in the old repo which didn't get any attention).

> Would be easier if related commits are in one PR to allow travis to build the packages.

OK, I thought I would stick to the contribution guideline but I've overlooked the part about "dependencies which are not part of void-packages yet." I can fix that.

> In general it would be nicer to build the wrapper from source as its gpl licensed.

I could include the patched version of the wrapper but since it is already is part of the archive which needs to be downloaded either way, I thought it would be easier/better to include the patch.

> The other part is proprietary and might not allow redistributing binaries, which would require `restricted=yes` to not add the package to the repository.
> In both cases allowed or not allowed to distribute, the package should be in the nonfree repository.

Redistribution (and modification) is allowed by Brothers license. I will add the nonfree parameter to the templates.

> Another issue would be depending on 32bit packages, this is not something void does.
> 32bit only packages should only build for `i686` and then `xbps-src` creates a `-32bit` package from with with the right dependencies.

I used the brother-dcp197-lpr packages template as reference, which also does this. Therefore I thought it was OK. But I will see if I can fix that.

File: srcpkgs/brother-mfc5490cn-cupswrapper/REMOVE [hunk 1]
```diff
@@ -1,3 +1,6 @@
 case ${ACTION} in
-    pre) /opt/brother/Printers/mfc5490cn/cupswrapper/cupswrappermfc5490cn -e ;;
+    pre)
+    	opt/brother/Printers/mfc5490cn/cupswrapper/cupswrappermfc5490cn -e ;;
```
Comment (pullmoll) [inline srcpkgs/brother-mfc5490cn-cupswrapper/REMOVE]: The case ends here at the `;;` and won't execute the following two lines.

File: srcpkgs/brother-mfc5490cn-cupswrapper/template [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+revision=2
 short_desc="CUPS wrapper driver for the Brother MFC-5490CN printer/scanner"
 maintainer="Henner M. Kruse <github@hmkruse.de>"
-license="GPL-2"
+license="GPL-2.0-only"
 homepage="http://support.brother.com/g/b/index.aspx"
 distfiles="http://download.brother.com/welcome/dlf006158/mfc5490cncupswrapper-${version}-2.i386.deb"
 checksum=6ecf0d967bfd8a1f8e3957551126a119c7bd929fd16348da8e664a6f7e25940f
+repository="nonfree"
+only_for_archs="i686"
```
Comment (pullmoll) [inline srcpkgs/brother-mfc5490cn-cupswrapper/template]: Meanwhile this has changed and should be `archs="i686"`.

File: srcpkgs/brother-mfc5490cn-lpr/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+version=1.1.2
+revision=2
+short_desc="LPR driver for Brother MFC-5490CN printer/scanner"
+maintainer="Henner M. Kruse <github@hmkruse.de>"
+license="proprietary"
+homepage="http://support.brother.com/g/b/index.aspx"
+distfiles="http://download.brother.com/welcome/dlf006156/mfc5490cnlpr-${version}-2.i386.deb"
+checksum=4a361dea157994cd825dbeba529ff560b279a591643ce8a0e0b1acce72f0bac5
+repository="nonfree"
+only_for_archs="i686"
```
Comment (pullmoll) [inline srcpkgs/brother-mfc5490cn-lpr/template]: `archs="i686"`

File: srcpkgs/brother-mfc5490cn-lpr/template [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+lib32mode="full"
+depends="a2ps ghostscript"
+nopie=yes
+#mutable_files="/opt/brother/Printers/mfc5490cn/inf/brmfc5490cnrc"
```
Comment (pullmoll) [inline srcpkgs/brother-mfc5490cn-lpr/template]: Remove this line.

Comment (pullmoll) [reply srcpkgs/brother-mfc5490cn-lpr/template]: And please squash the xlint fixes into their respective commits. This PR should have 2 commits.

**Comment by Gottox** [comment]
stalled for >6 months. Please reopen once there's progress.

**Pull request closed without merging**